src/MessageMap.ts

```diff
@@ -74,19 +101,26 @@ export class MessageMap {
 	}
 
 	getCorrespondingReverse(_direction: string, bridge: Bridge, toId: string) {
-		// The ID to return
-		let fromId = [];
-
-		// Get the mappings for this bridge
-		const keyToIdsMap = this._map.get(bridge);
-		if (keyToIdsMap !== undefined) {
-			// Find the ID
-			const [key] = [...keyToIdsMap].find(([, ids]) => ids.has(toId));
-			fromId = key.split(" ");
-			fromId.shift();
+		try {
+			// The ID to return
+			let fromId = [];
+
+			// Get the mappings for this bridge
+			const keyToIdsMap = this._map.get(bridge.name);
+			if (keyToIdsMap !== undefined) {
+				// Find the ID
+				const [key] = [...keyToIdsMap].find(([, ids]) => ids.has(toId));
```

Comment:
keyToIdsMap should be a map, right? So I think you need:
```suggestion
				const [key] = [...keyToIdsMap.values()].find(([, ids]) => ids.has(toId));
```

Reply:
This didn't work, actually broke the function for D -> T reverse key finding

Reply:
still in discussion?

Reply:
> still in discussion?

Yes, unable to make getCorrespondingReverse work, was planning to rework the entire MessageMap structure into sqlite when I have the time, something isn't working right here

Reply:
I got it working

Reply:
![image](https://user-images.githubusercontent.com/14029133/188467379-29572e41-f109-451e-a6a8-9b87aff753e7.png)